client/deployments/client.go

```diff
@@ -49,15 +88,92 @@ func NewClient(url string, skipVerify bool) *Client {
 
 	return &Client{
 		url:               url,
-		artifactUploadUrl: JoinURL(url, artifactUploadUrl),
+		artifactUploadURL: JoinURL(url, artifactUploadURL),
+		artifactsListURL:  JoinURL(url, artifactsListURL),
 		client: &http.Client{
 			Transport: tr,
 		},
 	}
 }
 
-func (c *Client) UploadArtifact(description, artifactPath, tokenPath string, noProgress bool) error {
+func (c *Client) ListArtifacts(tokenPath string, detailLevel int) error {
+	token, err := ioutil.ReadFile(tokenPath)
+	if err != nil {
+		return errors.Wrap(err, "Please Login first")
+	}
+
+	req, err := http.NewRequest(http.MethodGet, c.artifactsListURL, nil)
+	if err != nil {
+		return errors.Wrap(err, "Cannot create request")
+	}
+	req.Header.Set("Authorization", "Bearer "+string(token))
+
+	reqDump, _ := httputil.DumpRequest(req, false)
+	log.Verbf("sending request: \n%v", string(reqDump))
+
+	rsp, err := c.client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "Get /deployments/artifacts request failed")
+	}
+	defer rsp.Body.Close()
+
+	body, err := ioutil.ReadAll(rsp.Body)
+	if err != nil {
+		return err
+	}
+
+	var list artifactsList
+	err = json.Unmarshal(body, &list.artifacts)
+	if err != nil {
+		return err
+	}
+	for _, v := range list.artifacts {
+		listArtifact(v, detailLevel)
+	}
+
+	return nil
+}
+
+func listArtifact(a artifactData, detailLevel int) {
+	fmt.Println(fmt.Sprintf("ID: %s", a.ID))
+	fmt.Println(fmt.Sprintf("Name: %s", a.Name))
+	fmt.Println(fmt.Sprintf("Signed: %t", a.Signed))
+	fmt.Println(fmt.Sprintf("Modfied: %s", a.Modified))
+	fmt.Println(fmt.Sprintf("Size: %d", a.Size))
+	fmt.Println(fmt.Sprintf("Description: %s", a.Description))
+	if detailLevel == 1 {
```

Comment:
Question:
Why do we not want to list this at detail-level `2`?

Reply:
I'm fine with that. will move them to level 2